Makefile

```diff
@@ -50,3 +71,36 @@ cockpit/build: cockpit/download
 
 .PHONY: cockpit/devel
 cockpit/devel: cockpit/devel-uninstall cockpit/build cockpit/devel-install
+
+#
+# Building packages
+#
+
+RPM_SPEC=cockpit/$(PACKAGE_NAME).spec
+RPM_TARBALL=rpmbuild/SOURCES/$(PACKAGE_NAME)-$(VERSION).tar.gz
+NODE_MODULES_TEST=package-lock.json
+
+$(RPM_SPEC): $(RPM_SPEC) $(NODE_MODULES_TEST)
+	provides=$$(npm ls --omit dev --package-lock-only --depth=Infinity | grep -Eo '[^[:space:]]+@[^[:space:]]+' | sort -u | sed 's/^/Provides: bundled(npm(/; s/\(.*\)@/\1)) = /'); \
+	awk -v p="$$provides" '{gsub(/%{VERSION}/, "$(VERSION)"); $(SUB_NODE_ENV) gsub(/%{NPM_PROVIDES}/, p)}1' $< > $@
+
+$(RPM_TARBALL): export NODE_ENV ?= production
```

Comment:
something is wrong here I think, if NODE_ENV is set to production `npm ci` doesn't install the dev dependencies. I think NODE_ENV should be set to production after npm ci is run, but before the build is run

Reply:
We could just remove `npm ci` from the `cockpit/build` target.

Reply:
Oh right good catch

Reply:
Okay done